docs/components/top_app_bars.md

```diff
@@ -0,0 +1,346 @@
+iconId:
+path: /catalog/material-top-app-bar/
+-->
+
+**Instructions to authors:**
+* _Follow the instructions in each section &ndash; see the [accompanying examples](button-examples) for futher guidance._
+* _Delete these instructions before submitting your document_
+
+# Top app bar
+
+The [top app bar](https://material.io/components/app-bars-top/#) displays information and actions relating to the current screen.
+
+There are two types of top app bar:
+
+1. [Regular top app bar](#regular-top-app-bar)
+1. [Contextual action bar](#contextual-top-app-bar)
+<br>
+
+![Regular app bar: purple background, white text and icons and Contextual action bar: black background, white text and icons](assets/top_app_bars/Top-app-bars_types_side-by-side.png)
+
+## Using the top app bar
+
+The top app bar provides content and actions related to the current screen. It’s used for branding, screen titles, navigation, and actions.
+
+A regular top app bar can transform into a contextual action bar.
+
+Before you can use Material buttons, you need to import the Material Components package for Flutter: 
+```dart
+package:flutter/material.dart
+```
+
+You need to use [`MaterialApp`](https://api.flutter.dev/flutter/material/MaterialApp-class.html).
+
+For more information on getting started with the Material for Flutter, go to the Flutter [Material library](https://api.flutter.dev/flutter/material/material-library.html) page.
+
+
+### Making the top app bar accessible
+
+_List any accessiblity setting or attributes (such as labels), describe how to use them and link to any guidelines on the m.io site (for example, [how to write a good accessibility label for your component](https://material.io/design/usability/accessibility.html#writing))_
+
+
+
+## Regular top app bar
+
+The top app bar provides content and actions related to the current screen. It’s used for branding, screen titles, navigation, and actions.
+
+### Regular top app bar example
+
+`AppBar`
+* [Class definition](https://api.flutter.dev/flutter/material/AppBar-class.html)
+* [GitHub source](https://github.com/flutter/flutter/blob/master/packages/flutter/lib/src/material/app_bar.dart)
+* [Dartpad Demo](https://dartpad.dev/embed-flutter.html?gh_owner=material-components&gh_repo=material-components-flutter&gh_path=docs/components/dartpad/top-app-bar/regular&gh_ref=develop)
+
+The following example shows a top app bar with a page title, a navigation icon, two action icons, and an overflow menu:
+
+![Top app bar with nav icon, page title, favorite icon, and search icon](assets/top_app_bars/top_app_bar_screenshot.png)
+
+```dart
+AppBar(
+    leading: Icon(Icons.menu),
+    title: Text('Page title'),
+    actions: <Widget>[
+        Icon(Icons.favorite),
+        Padding(
+            padding: EdgeInsets.symmetric(horizontal: 16),
+            child: Icon(Icons.search),
+        ),
+        Icon(Icons.more_vert),
+    ],
+    backgroundColor: Colors.purple,
+),
+
+```
```

Comment:
nit: using 4 spaces instead of 2 compared to the theme example below

Reply:
Done